stdlib/public/Concurrency/ExecutorAssertions.swift

```diff
@@ -57,6 +57,8 @@ extension SerialExecutor {
       return
     }
 
+    // TODO: make message record and use checkOnExecutor with it and "crash" flag
```

Comment:
We can't do this easily without making the `_check` not SPI... Which I'd be okey with but we should chat about it.

Reply:
More context: we're not going to risk that change for 6.0, but we can do it on main afterwards